common-custom-user-data-gradle-plugin/src/main/java/com/gradle/CommonCustomUserDataGradlePlugin.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * This Java source file was generated by the Gradle 'init' task.
+ */
+package com.gradle;
+
+import com.gradle.enterprise.gradleplugin.GradleEnterpriseExtension;
+import com.gradle.enterprise.gradleplugin.GradleEnterprisePlugin;
+import com.gradle.scan.plugin.BuildScanExtension;
+import com.gradle.scan.plugin.BuildScanPlugin;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.initialization.Settings;
+
+public class CommonCustomUserDataGradlePlugin implements Plugin<Object> {
+    public void apply(Object target) {
+        if (target instanceof Settings) {
+            applySettingsPlugin((Settings) target);
+        } else if (target instanceof Project) {
+            applyProjectPlugin((Project) target);
+        }
+    }
+
+    private void applyProjectPlugin(Project project) {
+        project.getPlugins().withType(BuildScanPlugin.class, __ -> {
+            BuildScanExtension buildScan = project.getExtensions().getByType(GradleEnterpriseExtension.class).getBuildScan();
+            enhanceBuildScan(buildScan, project.getRootProject());
+        });
+    }
+
+    private void applySettingsPlugin(Settings settings) {
+        settings.getPlugins().withType(GradleEnterprisePlugin.class, __ ->
+                settings.getGradle().projectsLoaded(gradle -> {
```

Comment:
Not sure, do we have to wait for the projects being loaded?

Reply:
This was required to allow us to get project properties, which are used to detect Android Studio invocation. I'll see if there's another way...